client/src/luigi-client.js

```diff
@@ -187,394 +166,46 @@ const LuigiClient = {
-       * @param {('info'|'success'|'warning'|'error')} [settings.type] sets the type of the alert
-       * @param {Object} [settings.links] provides links data
-       * @param {Object} settings.links.LINK_KEY object containing the data for a particular link. To properly render the link in the alert message refer to the description of the **settings.text** parameter
-       * @param {string} settings.links.LINK_KEY.text text which replaces the link identifier in the alert content
-       * @param {string} settings.links.LINK_KEY.url url to navigate when you click the link. Currently, only internal links are supported in the form of relative or absolute paths.
-       * @param {number} settings.closeAfter (optional) time in milliseconds that tells Luigi when to close the Alert automatically. If not provided, the Alert will stay on until closed manually. It has to be greater than `100`.
-       * @returns {promise} which is resolved when the alert is dismissed. 
-       * @example
-       * import LuigiClient from '@kyma-project/luigi-client';
-       * const settings = {
-       *  text: Ut enim ad minim veniam, {goToHome} quis nostrud exercitation ullamco {relativePath} laboris nisi ut aliquip ex ea commodo consequat.
-       *    Duis aute irure dolor {goToOtherProject},
-       *  type: 'info',
-       *  links: {
-       *    goToHome: { text: 'homepage', url: '/overview' },
-       *    goToOtherProject: { text: 'other project', url: '/projects/pr2' },
-       *    relativePath: { text: 'relative hide side nav', url: 'hideSideNav' }
-       *  },
-       * closeAfter: 3000
-       * }
-       * LuigiClient
-       *  .uxManager()
-       *  .showAlert(settings)
-       *  .then(() => {
-       *     // Logic to execute when the alert is dismissed
-       * });
-
-       */
-      showAlert: function showAlert(settings) {
-        //generate random ID
-        settings.id = crypto.randomBytes(4).toString('hex');
-
-        if (settings.closeAfter && settings.closeAfter < 100) {
-          console.warn(
-            `Message with id='${
-              settings.id
-            }' has too small 'closeAfter' value. It needs to be at least 100ms.`
-          );
-          settings.closeAfter = undefined;
-        }
-
-        window.parent.postMessage(
-          {
-            msg: 'luigi.ux.alert.show',
-            data: {
-              settings
-            }
-          },
-          '*'
-        );
-
-        promises.alerts[settings.id] = {};
-        promises.alerts[settings.id].promise = new Promise(resolve => {
-          promises.alerts[settings.id].resolveFn = resolve;
-        });
-        return promises.alerts[settings.id].promise;
-      }
-    };
-  }
+  uxManager: () => uxManagerInstance
 };
 export default LuigiClient;
```

Comment:
As discussed, we could try to make it a class and export here a unique instance

Reply:
done